### PR TITLE
Add endpoint for registering seeds from remote datacenters

### PR DESCRIPTION
**Pull request by mrbrowning**

### CODE REVIEW

**Comment by benclarkwood** [comment]
retest this please

**Comment by benclarkwood** [comment]
@sascala Can you audit the C* docs for any references to the removed seed config parameters?

**Comment by benclarkwood** [comment]
(remote_seeds changed to remote_datacenter_urls)

File: frameworks/cassandra/build-cassandra-bin.bash [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+LIB_DIR="${CASSANDRA_DIST_NAME}/lib"
+_sha1sum "${LIB_DIR}"/*.jar &> libdir-before.txt || true
+
+echo "##### Install custom seed provider #####"
+cp -v ../seed-provider/build/libs/cassandra-seed-provider-${SEED_PROVIDER_VERSION}.jar ${LIB_DIR}
+
+echo "##### Install StatsD metrics support #####"
+
+# StatsD Reporter: custom build with newer version of metrics-core
```
Comment (nickbp) [inline frameworks/cassandra/build-cassandra-bin.bash]: This was written roughly a year ago, does this library still need to be custom built or is a stock build available now?

File: frameworks/cassandra/src/main/java/com/mesosphere/sdk/cassandra/api/Seeds.java [hunk 2]
```diff
@@ -12,10 +12,15 @@
...
     @JsonProperty("seeds")
     private final List<String> seeds;
+    @JsonProperty("is_seed")
+    private final boolean isSeed;
 
     @JsonCreator
-    public Seeds(@JsonProperty("seeds") List<String> seeds) {
+    public Seeds(
+            @JsonProperty("seeds") List<String> seeds,
+            @JsonProperty("is_seed") boolean isSeed) {
```
Comment (nickbp) [inline frameworks/cassandra/src/main/java/com/mesosphere/sdk/cassandra/api/Seeds.java]: What does this value mean?

File: frameworks/cassandra/src/main/java/com/mesosphere/sdk/cassandra/scheduler/Main.java [hunk 3]
```diff
@@ -38,35 +25,21 @@ public static void main(String[] args) throws Exception {
...
-    private static Collection<Object> getResources(List<String> localSeeds) {
-        Collection<String> configuredSeeds = new ArrayList<>(localSeeds);
-
-        String remoteSeeds = System.getenv("TASKCFG_ALL_REMOTE_SEEDS");
-        if (!StringUtils.isEmpty(remoteSeeds)) {
-            configuredSeeds.addAll(Arrays.asList(remoteSeeds.split(",")));
-        }
+    private static Collection<Object> getResources(StateStore stateStore) {
+        final Collection<Object> apiResources = new ArrayList<>();
+        apiResources.add(new SeedsResource(stateStore, 2));
```
Comment (nickbp) [inline frameworks/cassandra/src/main/java/com/mesosphere/sdk/cassandra/scheduler/Main.java]: Should this be using config.json's seed_count setting? Currently hardcoded..

File: frameworks/cassandra/src/main/java/com/mesosphere/sdk/cassandra/scheduler/Main.java [hunk 4]
```diff
@@ -38,35 +25,21 @@ public static void main(String[] args) throws Exception {
...
                 DefaultServiceSpec.newGenerator(rawServiceSpec, schedulerFlags)
-                        .setAllPodsEnv("LOCAL_SEEDS", Joiner.on(',').join(localSeeds))
                         .build(),
                 schedulerFlags)
                 .setPlansFrom(rawServiceSpec)
-                .setCustomResources(getResources(localSeeds))
                 .setRecoveryManagerFactory(new CassandraRecoveryPlanOverriderFactory());
-        return schedulerBuilder;
-    }
+        schedulerBuilder.setCustomResources(getResources(schedulerBuilder.getStateStore()));
```
Comment (nickbp) [inline frameworks/cassandra/src/main/java/com/mesosphere/sdk/cassandra/scheduler/Main.java]: Should continue using the builder chain IMO, could even just return the result of the chain without a `schedulerBuilder` variable

File: frameworks/cassandra/universe/marathon.json.mustache [hunk 5]
```diff
@@ -129,6 +129,7 @@
     "TASKCFG_ALL_CASSANDRA_STREAMING_SOCKET_TIMEOUT_IN_MS": "{{cassandra.streaming_socket_timeout_in_ms}}",
     "TASKCFG_ALL_CASSANDRA_MAX_VALUE_SIZE_IN_MB": "{{cassandra.max_value_size_in_mb}}",
     "TASKCFG_ALL_CASSANDRA_OTC_COALESCING_STRATEGY": "{{cassandra.otc_coalescing_strategy}}",
+    "TASKCFG_ALL_LOCAL_SEEDS": "node-0-server.{{service.name}}.autoip.dcos.thisdcos.directory,node-1-server.{{service.name}}.autoip.dcos.thisdcos.directory",
```
Comment (nickbp) [inline frameworks/cassandra/universe/marathon.json.mustache]: This similarly ignores the seed_count setting. I think this is what the logic in Main.java was for.

**Pull request closed without merging**